associations.rb

```diff
@@ -0,0 +1,52 @@
+class Classroom
+  attr_accessor :label, :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students << student
+    student.classroom = self
+  end
+end
+
+class Student
+  attr_accessor :name, :classroom
+
+  def initialize(name)
+    @name = name
+  end
+end
+
+class Rental
+  attr_accessor :book, :person, :date
+
+  def initialize(book, person, date)
+    @book = book
+    @person = person
+    @date = date
+    @book.rentals << self
+    @person.rentals << self
+  end
+end
+
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```

Comment:
- [ ] Kindy make sure `You create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental` as said in the requirements. A method to add a rental should be implemented here in the book and you must make sure it has `person`, `date` and the book its self.

Hint: 
```rb
def add_rental(person, date)
    Rental.new(date, self, person)
  end
```